### PR TITLE
Fix splitter with banned sites

### PR DESCRIPTION
**Pull request by mesmith75**

### CODE REVIEW

File: ganga/GangaDirac/Lib/Splitters/OfflineGangaDiracSplitter.py [hunk 1]
```diff
@@ -257,7 +257,7 @@ def calculateSiteSEMapping(file_replicas, uniqueSE, CE_to_SE_mapping, SE_to_CE_m
         SE_dict[lfn] = sitez
 
     # Remove the banned sites (CE) from the mappings
-    for iSE in CE_to_SE_mapping.keys():
+    for iSE in list(CE_to_SE_mapping.keys()):
```
Comment (alexanderrichards) [inline ganga/GangaDirac/Lib/Splitters/OfflineGangaDiracSplitter.py:260]: Why is this necessary? `.keys()` should return an iterable generator.

Could also do the following:
```suggestion
    for iSE in CE_to_SE_mapping:
```

**Review by alexanderrichards — CHANGES_REQUESTED** [review]

File: ganga/GangaDirac/Lib/Splitters/OfflineGangaDiracSplitter.py (hunk 1 above)
Comment (mesmith75) [reply ganga/GangaDirac/Lib/Splitters/OfflineGangaDiracSplitter.py:260]: But you need a copy of the list if you start to remove items. Other wise you end up with
```python
RuntimeError: dictionary changed size during iteration
```

**Review by egede — APPROVED** [review]

File: ganga/GangaDirac/Lib/Splitters/OfflineGangaDiracSplitter.py (hunk 1 above)
Comment (alexanderrichards) [reply ganga/GangaDirac/Lib/Splitters/OfflineGangaDiracSplitter.py:260]: Ah sure, I didn't read the following code, just this line.

**Review by alexanderrichards — APPROVED** [review]
